src/EditorFeatures/Test/Completion/AbstractCompletionProviderTests.cs

```diff
@@ -58,6 +58,12 @@ internal CompletionServiceWithProviders GetCompletionService(Workspace workspace
         internal abstract CompletionServiceWithProviders CreateCompletionService(
             Workspace workspace, ImmutableArray<CompletionProvider> exclusiveProviders);
 
+        protected abstract string ItemPartiallyWritten(string expectedItemOrNull);
+
+        protected abstract Task<TestWorkspace> CreateWorkspaceAsync(string fileContents);
+
+        protected abstract Task BaseVerifyWorkerAsync(string code, int position, string expectedItemOrNull, string expectedDescriptionOrNull, SourceCodeKind sourceCodeKind, bool usePreviousCharAsTrigger, bool checkForAbsence, int? glyph);
```

Comment:
i don't like how this test class works *at all*.  It has baked in logic, then has classes that derive from that and specialize, and then there are even deeper subclasses that then do *not* want that specialization.  It's basically 100% broken.